src/main/java/io/myprojects/tweetproject/MyTweetService.java

```diff
@@ -0,0 +1,200 @@
+		tweetStat = new TweetStatisticBuilder()
+				.setSumOfLikes(sumOfLikes)
+				.setLikesAverage(likesAvg)
+				.setLikesMedian(likesMedian)
+				.setLikesPercentiles(likesPercentiles)
+				.setTweetCount(tweetMap.size())
+				.setMaxNumberOfLikes(maxNumberOfLikes)
+				.setMinNumberOfLikes(minNumberOfLikes)
+				.setMostLikedTweet(mostLikedTweet)
+				.build();
+		return tweetStat;
+	}
+	
+	private Tweet getMostLikedTweet(List<Tweet> localTweetArray) {		
+		return getTweet(localTweetArray.stream().max(comp).get().getTweetId());
+	}
+
+	private int getMaxNumberOfLikes(List<Tweet> localTweetArray) {
+		return localTweetArray.stream().max(comp).get().getNumOfLikes();
+	}
+
+	private int getMinNumberOfLikes(List<Tweet> localTweetArray) {
+		return localTweetArray.stream().min(comp).get().getNumOfLikes();
+	}
+
+	private double getAvgOfLikes(Map<Integer, Tweet> localTweetMap, List<Tweet> localTweetArray) {
+		return localTweetArray.stream().mapToDouble(l -> l.getNumOfLikes()).average().getAsDouble();
+	}
+	
+	private long getSumOfLikes(List<Tweet> localTweetArray) {
+		return localTweetArray.stream().mapToLong(l -> l.getNumOfLikes()).sum();
+	}
+
+	private long getLikesMedian(List<Tweet> localTweetArray) {
+		long medianValue;
+		Integer [] likesArray = getLikesArrayFromTweetList(localTweetArray);
+		Arrays.sort(likesArray);
+		int median = likesArray.length /2;
+		if (likesArray.length % 2 == 1) {
+			medianValue = (long)likesArray[median];
+			return medianValue;
+		} else {
+			medianValue = (long)(likesArray[median - 1] + likesArray[median]) / 2;
+			return medianValue;
+		}
+	}
+
+	private Integer[] getLikesArrayFromTweetList(List<Tweet> tweetList) {		
+		return tweetList.stream().map(tweet-> tweet.getNumOfLikes()).toArray(Integer[]::new);
+	}
+
+	private int[] getLikesPercentiles(List<Tweet> localTweetArray) {		
+		Integer [] likesArray = getLikesArrayFromTweetList(localTweetArray);
+		Double[] likesPercentiles = getDoubleArray(likesPercentileArray);
+		int[] likesPercentilesResult = percentiles(likesArray, likesPercentiles);
+		return likesPercentilesResult;	
+	}
+
+	private Double[] getDoubleArray(String[] likesPercentileArray) {
+		Double[] likesPercentilesDoubleArr = Arrays.stream(likesPercentileArray).map(Double::valueOf).toArray(Double[]::new);
+		return Arrays.stream(likesPercentilesDoubleArr).map(p->p/100).toArray(Double[]::new);		
+	}
+
+	private static int[] percentiles(Integer[] latencies, Double... percentiles) {
+		Arrays.sort(latencies, 0, latencies.length);
+		int[] values = new int[percentiles.length];
+		for (int i = 0; i < percentiles.length; i++) {
+			int index = (int) (percentiles[i] * latencies.length);
+			values[i] = latencies[index];
+		}
+		return values;
+	}
+
+	public TweetStatistic getStats(int userId) {
+		 Map<Integer,Tweet> tempTweetMap = new HashMap<Integer,Tweet>();
+		 List<Tweet> tempTweetArray = new ArrayList<Tweet>();
+		 
+		for(Tweet tweet : tweetArray) {

```

Comment:
With such a pattern of operations is classic to use java8 streams
For example:
`List<Tweet> userTweets = tweetArray.stream().filter(x -> userId == x.getUserId()).collect(Collectors.toList());`

And if you want a Map (as I said in earlier comment, you don't actually need the array)
you can go with the following:
```
Map<Integer, Tweet> userTweetsMap = tweetArray.stream().filter(x -> userId == x.getUserId())
.collect(Collectors.toMap(Tweet::getTweetId, Function.identity()));

```